internal/balancergroup/balancergroup.go

```diff
@@ -332,6 +338,31 @@ func (bg *BalancerGroup) Add(id string, builder balancer.Builder) {
 	bg.outgoingMu.Unlock()
 }
 
+// UpdateBuilder updates the builder for a current child, starting the Graceful
+// Switch process for that child.
+func (bg *BalancerGroup) UpdateBuilder(id string, builder balancer.Builder) {
+	bg.outgoingMu.Lock()
+	// This does not deal with the balancer cache because this call should come
+	// after an Add call for a given child balancer. If the child is removed,
+	// the caller will call Add if the child balancer comes back which would
+	// then deal with the balancer cache.
+	sbc := bg.idToBalancerConfig[id]
+	if sbc == nil {
+		// simply ignore it if not present, don't error
+		return
+	}
+	sbc.builder = builder
+	// Even if you get an add and it persists builder but doesn't start
+	// balancer, this would leave graceful switch being nil, in which we are
+	// correctly overwriting with the recent builder here as well to use later.
+	// The graceful switch balancer's presence is an invariant of whether the
+	// balancer group is closed or not (if closed, nil, if started, present).
+	if sbc.balancer != nil {
+		sbc.gracefulSwitch()
+	}
```

Comment:
Why not make `sbc.gracefulSwitch()` take the builder as the input? Then it can update the field, and also do the `nil` check.

Reply:
The reason I prefer this is that, `sbc.balancer` should ideally dealt with as a private field of the subBalancerWrapper. It should only be read by the subBalancerWrapper methods, not outside.
And with the way it is, it's like `sbc.gracefulSwitch` only does half of the job.

The code is not updated. Did you not push? Or haven't changed it yet?